build/build.go

```diff
@@ -889,3 +891,34 @@ func (w *waitingWriter) Close() error {
 	}
 	return err
 }
+
+// handle https://github.com/moby/moby/pull/10858
+func handleLowercaseDockerfile(dir, p string) string {
+	if filepath.Base(p) != "Dockerfile" {
+		return p
+	}
+
+	f, err := os.Open(filepath.Dir(filepath.Join(dir, p)))
+	if err != nil {
+		return p
+	}
+
+	names, err := f.Readdirnames(-1)
```

Comment:
Was wondering why you were using `f.Readdirnames` here, but then realised`os.Stat()` returns the name you pass to it, not the actual filename on disk (preserving case) 😞 

Wondering, if `Readdirnames()` would be an issue if there's lots of files in the directory.

Would something like this work?

```go
dockerfiles, err := filepath.Glob(p + "/[Dd]ockerfile")
```

I did a quick test on my machine;

```go
package main

import (
	"fmt"
	"path/filepath"
)

func main() {
	dockerfileDir, _ := filepath.Abs(".")
	dockerfileName := "Dockerfile"
	f := handleLowercaseDockerfile(dockerfileDir, dockerfileName)
	fmt.Println("Using:", f)
}

// handle https://github.com/moby/moby/pull/10858
func handleLowercaseDockerfile(dir, p string) string {
	if filepath.Base(p) != "Dockerfile" {
		return p
	}
	dockerfiles, err := filepath.Glob(filepath.Join(dir, "[Dd]ockerfile"))
	if err != nil || len(dockerfiles) != 1 {
		return p
	}
	// either Dockerfile or dockerfile was found; use what's found
	return filepath.Base(dockerfiles[0])
}
```

```bash
rm -f Dockerfile && touch dOckErFILE && go run main.go
rm -f Dockerfile && touch Dockerfile && go run main.go
rm -f Dockerfile && touch dockerfile && go run main.go

Using: Dockerfile
Using: Dockerfile
Using: dockerfile
```

Reply:
`Glob` is implemented with `Readdirnames(-1)`

Reply:
sigh 😔

it really surprised me that os.Stat doesn't return the actual filename